apps/avifenc.c

```diff
@@ -2009,8 +2009,22 @@ int main(int argc, char * argv[])
             fprintf(stderr, "rav1e doesn't support lossless encoding yet: https://github.com/xiph/rav1e/issues/151\n");
             goto cleanup;
         } else if (codecName && !strcmp(codecName, "svt")) {
-            fprintf(stderr, "SVT-AV1 doesn't support lossless encoding yet: https://gitlab.com/AOMediaCodec/SVT-AV1/-/issues/1636\n");
-            goto cleanup;
+            char versions[256];
+            avifCodecVersions(versions);
+            const char svtVersionPrefix[] = "svt [enc]:v";
+            const char * svtVersionPrefixPos = strstr(versions, svtVersionPrefix);
+            if (svtVersionPrefixPos == NULL) {
+                // Make sure the syntax returned by avifCodecVersions() did not change.
+                assert(strstr(versions, "svt") == NULL && strstr(versions, "SVT") == NULL);
+                // Let the encode fail later because SVT was not included in the build.
```

Comment:
Optional: We probably should return an error here, because the libavif library is apparently malfunctioning.

Reply:
It might just be `avifCodecVersions()` that returns another string, for example "svt-av1 [enc]:v1.2.3" instead of the expected "svt [enc]:v1.2.3". Catching that in the CI with an assertion is likely enough, no need to be stricter as the only side effect is allowing lossy encoding with -q 100 for SVT < 3.0.0.

Reply:
Also `avifenc -c svt --lossless` fails with "Matrix coefficients have to be identity, YCgCo-Re, or YCgCo-Ro in lossless mode".  
`avifenc -c svt --lossless --cicp x/x/0` fails with "matrixCoefficients may not be set to identity (0) when subsampling".  
So only `--cicp x/x/16" works which is quite niche.

Reply:
These two error messages are correct. They are confusing because it is the default matrix coefficients or chroma subsampling format that causes the error. I guess we can clarify them by printing the matrix coefficients and the chroma subsampling format.

Reply:
The former is guarded by `settings.cicpExplicitlySet` so it is not confusing.

The latter is mentioning `matrixCoefficients may not be set to identity (0) when subsampling` so stating 4:2:0 or 4:2:2 does not bring much. Monochrome is checked above.